clang/lib/Sema/SemaType.cpp

```diff
@@ -8086,23 +8086,21 @@ static void HandleNeonVectorTypeAttr(QualType &CurType, const ParsedAttr &Attr,
 
   // Target must have NEON (or MVE, whose vectors are similar enough
   // not to need a separate attribute)
-  if (!(S.Context.getTargetInfo().hasFeature("neon") ||
-        S.Context.getTargetInfo().hasFeature("mve") ||
-        S.Context.getTargetInfo().hasFeature("sve") ||
-        S.Context.getTargetInfo().hasFeature("sme") ||
+  if (!(S.Context.getTargetInfo().hasFeature("mve") ||
         IsTargetCUDAAndHostARM) &&
```

Comment:
it looks to me like the logic here can be simplified and use of this variable can be removed entirely.

After reading 
https://github.com/llvm/llvm-project/commit/8b0ea48740935d819618d8254fc45d98179b672c

It looks to me like somebody already disabled these errors when targeting GPUs, so we could just simplify the logic and emit the errors only when user targets M class CPUs, what do you think?

Reply:
I am suggesting that in that patch we removed error emission when targeting GPU, since now we are restricting error emission to M class processors, I think the extra handling for GPUS can be removed.

Reply:
Ah now I understand. What you are trying to see that for mve there is no reason to check for GPUs as they will never be target for this extension?